src/models/chessboard.py

```
@@ -68,8 +124,83 @@ async def get_poses(
         timeout: Optional[float] = None,
         **kwargs
     ) -> Dict[str, PoseInFrame]:
-        self.logger.error("`get_poses` is not implemented")
-        raise NotImplementedError()
+        
+        cam_images = await self.camera.get_images()
+        for image in cam_images[0]:
+            # TODO: Check if we should only receive JPEG images. I feel like we should expand to any image format
+            if image.mime_type == CameraMimeType.JPEG:
+                viam_image = cam_images[0][0].data
+                self.logger.debug("Found image from camera")
+        if viam_image is None:
+            err = "Could not get latest image from camera"
+            self.logger.error(err)
+            return err, None
+        
+        pil_image = viam_to_pil_image(viam_image)
+        image = np.array(pil_image)
+        
+        # Convert to grayscale if needed
+        if len(image.shape) == 3:
+            image = cv2.cvtColor(image, cv2.COLOR_RGB2GRAY)
+        
+        K, dist = await self.get_camera_intrinsics()
+
+        found, corners = cv2.findChessboardCorners(
+            image,
+            self.pattern_size,
+            flags=cv2.CALIB_CB_ADAPTIVE_THRESH + cv2.CALIB_CB_NORMALIZE_IMAGE
+        )
+        if not found:
+            err = "Could not find chessboard pattern in image"
+            self.logger.error(err)
+            return err, None
```

This seems like we're using go patterns in python. Should we just raise exceptions instead of returning these errors?